File: app-spring-endpoints/src/main/java/app/model/Car.java
package app.model;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "car")
public class Car implements Serializable {

	@Id
	@Column
	private int year;
	
	@Id
	@Column
	private String manufacturer;
	
	@Id
	@Column
	private String model;
	
	
	public int getYear() {
		return year;
	}
	public void setYear(int year) {
		this.year = year;
	}

	public String getManufacturer() {
		return manufacturer;
	}
	public void setManufacturer(String manufacturer) {
		this.manufacturer = manufacturer;
	}

	public String getModel() {
		return model;
	}
	public void setModel(String model) {
		this.model = model;
	}
	
	public String toString() {
		return "[year = " + this.year + "]" + 
				", [manufacturer = " + this.manufacturer + "]" +
				", [model = " + this.model + "]";
	}

}

File: app-redux-table/src/app/model/car.ts
export class Car {
	private _internalId: number;
	private _year: number;
	private _manufacturer: string;
	private _model: string;

	constructor(internalId: number, year: number, manufacturer: string, model: string) {
		this._internalId = internalId;
		this._year = year;
		this._manufacturer = manufacturer;
		this._model = model;
	}

	public get internalId(): number {
		return this._internalId;
	}

	public get year(): number {
		return this._year;
	}

	public get manufacturer(): string {
		return this._manufacturer;
	}

	public get model(): string {
		return this._model;
	}
}